Added script to update `data` from genshin-optimizer

soleilvermeil opened the pull request:
At current stage, the code needs a `.env` file with fields `GITHUB_USERNAME` and `GITHUB_TOKEN` in order to be able to make up to 5000 requests to the GitHub API. Run the script, use `go2eiko.py -h` for more infos.

soleilvermeil committed:
Added script to update `data` from genshin-optimizer

At current stage, the code needs a `.env` file with fields `GITHUB_USERNAME` and `GITHUB_TOKEN` in order to be able to make up to 5000 requests to the GitHub API. Run the script, use `go2eiko.py -h` for more infos.

soleilvermeil commented:
Of course I meant `python go2eiko.py -h` instead of `go2eiko.py -h`...

eikofee commented:
*link to #27 for reference*

soleilvermeil committed:
Added cards retrieving

soleilvermeil committed:
Files can be renamed according to index.ts

If using `download_recursive`, the user can now chose to specify `"index.ts"` rather than a callable for renaming function, which will indicate that the program should rather look for files called `index.ts` to rename the files.

soleilvermeil committed:
Added eikonomiya's naming conventions

eikofee commented on go2eiko.py:
```diff
@@ -4,6 +4,7 @@
 import logging
 from dotenv import load_dotenv
```
dotenv will probably be removed to use "true" env variables, which would be easier for the user to setup using Docker.

soleilvermeil replied:
Changed in ed9428c16dbeea8794ec887d62dab078968c52fe (see also https://github.com/eikofee/eikonomiya/pull/28#discussion_r1554844413)

eikofee commented on go2eiko.py:
```diff
@@ -0,0 +1,336 @@
+        "constellation5": "c5",
+        "constellation6": "c6",
+        "icon": "face",
+        "banner": "namecard",
```
there is also characters cards in `gi/char-cards/src` : `Character_<name>_Card` -> `characters/name/card` but they are quite low-res so I'll probably look elsewhere.

eikofee commented on go2eiko.py:
```diff
@@ -0,0 +1,336 @@
+
+            # DEBUG
+            # if folder != "Furina":
+            #     continue
```
kek

eikofee commented on go2eiko.py:
```diff
@@ -0,0 +1,336 @@
+    dotenv_path = os.path.join(os.path.dirname(__file__), '.env')
+    load_dotenv(dotenv_path)
+    username = os.environ.get('GITHUB_USERNAME')
+    token = os.environ.get('GITHUB_TOKEN')
```
if we're not using dotenv later on, these variables should be named something like "EIKONOMIYA_GITHUB_STUFF"

soleilvermeil replied:
Changed in ed9428c16dbeea8794ec887d62dab078968c52fe (see also https://github.com/eikofee/eikonomiya/pull/28#discussion_r1554842576)

soleilvermeil committed:
Added an option to use git's sparse checkout

This new method, usable via `--method checkout` flag allows the program to clone genshin-optimizer's repo (partially), puts everything in `temp`. It then recovers files directly from there, without needing the use of GitHub's API.

soleilvermeil committed:
Removed need of a `.env` for sparse checkouts

eikofee commented on go2eiko.py:
```diff
@@ -5,11 +5,14 @@
-EXTENSIONS = ['.png', '.jpg', '.jpeg']
+EXTENSIONS = ['png', 'jpg', 'jpeg']
 AUTH = None  # TODO: might not be the most secured way to store the credentials
+TEMP_FOLDER = "temp"
```
could you change it to `data/temp` so it gets ignored by git and docker ?

eikofee commented on go2eiko.py:
```diff
@@ -0,0 +1,439 @@
+
+        # Check in which case we are
+        if "Awaken" in old_filename:
+            return f"weapon-awaken.{ext}"
```
eikonomiya uses `weapon-awake`, not `weapon-awaken`

soleilvermeil replied:
Fixed in ed9428c16dbeea8794ec887d62dab078968c52fe

eikofee commented on go2eiko.py:
```diff
@@ -0,0 +1,439 @@
+        repository, and return the name of the file as it should be saved.
+    """
+
+    if METHOD == "checkout":
```
is it possible to delete the temp folder after that ? it currently uses 190MB after the operation

soleilvermeil replied:
Working on it. Some files in read only are giving me headache

soleilvermeil replied:
Fixed in c61923c0fa68e435d494832de8b02fe3233665c7
You can now also use the flag `--keep` to avoid the deletion of the `temp` folder

eikofee commented:
it's almost good, but images of ascension 1 and 4 passives are not renamed yet, nor skill & burst.
finally, there's also common assets regarding auto-attacks depending on character weapon but i'm not really satisfied with the current organisation in eikonomiya about it so it's fine.

soleilvermeil committed:
Merge branch 'eikofee:master' into master

soleilvermeil committed:
Refactored env vars and folders

* Changed the names of the environment variables when `--method api` is used.
* When using `--method checkout`, data is now stored into `<output>/temp`, where `<output>` is given by `--output <output>`, default being `data`.
* Added images for skill, burst, passives for ascension 1 & 4
* Fixed the name for awakened weapons

soleilvermeil commented:
> it's almost good, but images of ascension 1 and 4 passives are not renamed yet, nor skill & burst. finally, there's also common assets regarding auto-attacks depending on character weapon but i'm not really satisfied with the current organisation in eikonomiya about it so it's fine.

Fixed in ed9428c16dbeea8794ec887d62dab078968c52fe for a1, a4, skill and burst

soleilvermeil committed:
Temp folder is deleted after execution
